server/src/main/java/org/opensearch/tasks/consumer/TaskDetailsLogMessage.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks.consumer;
+
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.common.logging.OpenSearchLogMessage;
+import org.opensearch.tasks.Task;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public final class TaskDetailsLogMessage extends OpenSearchLogMessage {
+    TaskDetailsLogMessage(Task task) {
+        super(prepareMap(task), message(task));
+    }
+
+    private static Map<String, Object> prepareMap(Task task) {
+        Map<String, Object> messageFields = new HashMap<>();
+        messageFields.put("taskId", task.getId());
+        messageFields.put("type", task.getType());
+        messageFields.put("action", task.getAction());
+        messageFields.put("description", task.getDescription());
+        messageFields.put("start_time_millis", TimeUnit.NANOSECONDS.toMillis(task.getStartTime()));
+        messageFields.put("parentTaskId", task.getParentTaskId());
+        messageFields.put("resource_stats", task.getResourceStats());
+        messageFields.put("metadata", task instanceof SearchShardTask ? ((SearchShardTask) task).getTaskMetadata() : null);
```

Comment:
This is telling me that `getTaskMetadata` needs to be a feature of `Task`, or `TaskDetailsLogMessage` needs to be specialized into a `SearchShardTaskDetailsLogMessage`.

Reply:
+1 for the latter, and have it accept a `SearchShardTask` instance instead of the `Task` superclass

Reply:
Done modified this to ```SearchShardTaskDetailsLogMessage``` and this class accepts ```SearchShardTask```.